### PR TITLE
student_registry modified

### PR DESCRIPTION
**Pull request by SamarthMayya**

### CODE REVIEW

File: session_2/student_registry/db/migrate/20210114031001_add_fields_to_students.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class AddFieldsToStudents < ActiveRecord::Migration[6.1]
+  def change
+    add_column :students, :branch, :string
+    add_column :students, :cgpa, :decimal, precision:2, scale: 2
```
Comment (abhishekkumar2718) [inline session_2/student_registry/db/migrate/20210114031001_add_fields_to_students.rb:4]: Check if you can store `10.00` as the CGPA - it's definitetly a valid cgpa, right?

**Review by abhishekkumar2718 — CHANGES_REQUESTED** [review]

File: session_2/student_registry/db/migrate/20210114031001_add_fields_to_students.rb (hunk 1 above)
Comment (SamarthMayya) [reply session_2/student_registry/db/migrate/20210114031001_add_fields_to_students.rb:4]: Precision is 2, it should be able to store right?

**Comment by SamarthMayya** [comment]
I'm done with session 2, could you please review the changes?

File: session_2/cricviz/app/models/cricketer.rb [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-  scope :batters, -> { raise NotImplementedError }
+  scope :batters, -> { where(role: 'Batter') }
 
   # Select players with the role 'Bowler'
-  scope :bowlers, -> { raise NotImplementedError }
+  scope :bowlers, -> { where(role: 'Bowler') }
 
   # Sort players by the descending number of matches played
-  scope :descending_by_matches, -> { raise NotImplementedError }
+  scope :descending_by_matches, -> { all.order(matches: :desc) }
```
Comment (abhishekkumar2718) [inline session_2/cricviz/app/models/cricketer.rb:12]: `all` can be left - it's implicit as with the other `where` scopes

File: session_2/cricviz/app/models/cricketer.rb [hunk 3]
```diff
@@ -68,14 +80,50 @@ def self.import_classical_batters
   # `ActiveRecord::RecordNotFound` exception with the player's name as
   # the message.
   def self.update_innings(batting_scorecard, bowling_scorecard)
-    raise NotImplementedError
+    batting_scorecard.each do |batting_scorecard1|
+      bat = Cricketer.find_by(name: batting_scorecard1[0])
+      if( !bat )
```
Comment (abhishekkumar2718) [inline session_2/cricviz/app/models/cricketer.rb:85]: Don't need brackets at all for simple expressions

File: session_2/cricviz/app/models/cricketer.rb [hunk 4]
```diff
@@ -19,7 +19,12 @@ class Cricketer < ApplicationRecord
   # - If the player has not batted yet, return nil
   # - If the player has been not out in all innings, return runs scored.
   def batting_average
-    raise NotImplementedError
+    if(runs_scored.nil? || innings_batted.nil? || not_out.nil? )
```
Comment (abhishekkumar2718) [inline session_2/cricviz/app/models/cricketer.rb:22]: Don't need brackets for simple expressions

File: session_2/student_registry/db/migrate/20210114031001_add_fields_to_students.rb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 class AddFieldsToStudents < ActiveRecord::Migration[6.1]
   def change
     add_column :students, :branch, :string
-    add_column :students, :cgpa, :decimal, precision:2, scale: 2
+    add_column :students, :cgpa, :decimal, precision:4, scale: 2
```
Comment (abhishekkumar2718) [inline session_2/student_registry/db/migrate/20210114031001_add_fields_to_students.rb:4]: With a precision of 4, I can save values like `15.00`

Can you write validations to ensure that CGPA is between 0 and 10?

https://guides.rubyonrails.org/active_record_validations.html

**Review by abhishekkumar2718 — CHANGES_REQUESTED** [review]

**Comment by abhishekkumar2718** [comment]
Hey!

Your PR looks fine apart from some style changes, great work :partying_face:

**Pull request closed without merging**